tests/mocks.py

```diff
@@ -1037,6 +1052,47 @@ def to_response(i):
 
         return to_response(diffs[0])
 
+    @conduit_method("transaction.search")
+    def transaction_search(
+        self,
+        *,
+        objectIdentifier=None,
+        constraints={},
+        order=None,
+        before=None,
+        after=None,
+        limit=100,
+    ):
+        matching_transactions = [
+            t for t in self._transactions if t["objectPHID"] == objectIdentifier
+        ]
+        # We don't want tests that work with our returned result to accidentally
+        # modify a transaction object.
+        matching_transactions = deepcopy(matching_transactions)
```

Comment:
Convention is to give every method a `to_response(i)` which deals with serializing an item. Fields are all explicitly copied so if anything needs to be added in the future (such as linking to other objects, this happened with edges in the past) you don't start spilling extra data into responses. It'd be good to explicitly fail in the case of a transaction type which we haven't added support for yet.
```suggestion
        def to_response(i):
            if i["type"] == "comment":
                return deepcopy(
                    {
                        "id": i["id"],
                        "type": i["type"],
                        "objectPHID": i["objectPHID"],
                        "comments": i["comments"],
                        # ... TODO: rest of required items here.
                    }
                )
            
            raise ValueError(
                "PhabricatorDouble transactions do not have support"
                'for the "{}" transaction type. '
                "If you have added use of a new transaction type please "
                "update PhabricatorDouble to support it.".format(i["type"])
            )
```

Reply:
I added the `to_response()` function but kept the type-checking code out of the method.  Transactions are so varied in structure that I think it makes sense for the method caller to have the freedom to put whatever they want in there.  I did this in the test code and it was nice to use.  Also it would take a long time to reverse engineer every transaction type, codify, then enforce its structure.  I don't see the benefit to writing that code at this early point with so few uses of the transaction code in the codebase.

Reply:
Sorry, to be clear I'm not suggesting you should type check against all the real transactions. What I mean is type checking for only the transactions that this mock can properly represent. The reason for this is so that if someone adds a new use of a transaction type to Lando the `PhabricatorDouble` won't silently represent it the wrong way, tricking the author into thinking things are working when they're using the wrong structure for that transaction type. If it turns out "TransactionA" is the same structure as "comment" lets say, then the dev implementing things will explicitly mark that and move on.